### PR TITLE
sratoolkit: 2.11.3 -> 3.1.1, build from source

### LINKED ISSUE #322164: Update request: sratoolkit 2.11.3 → 3.1.1
- Package name: sratoolkit
- Latest released version: 3.1.1
- Current version on the unstable channel: 2.11.3
- Current version on the stable/release channel: 2.11.3

- [ x ] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

**Notify maintainers**

@thyol

-----

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by t4ccer**
## Description of changes

Close #322164

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by t4ccer** [comment]
Result of `nixpkgs-review pr 326174` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>ncbi-vdb</li>
    <li>sratoolkit</li>
  </ul>
</details>

**Comment by idlip** [comment]
Hi @t4ccer, just to add, can you format those package files with `nixfmt-rfc-style`.

Thank you for pr, hope it gets merged soon.

**Comment by t4ccer** [comment]
Formatted!

File: pkgs/by-name/nc/ncbi-vdb/package.nix [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "ncbi-vdb";
+  version = "3.1.1";
+
+  src = fetchFromGitHub {
+    owner = "ncbi";
+    repo = "ncbi-vdb";
+    rev = finalAttrs.version;
```
Comment (gador) [inline pkgs/by-name/nc/ncbi-vdb/package.nix:20]:
```suggestion
    rev = "refs/tags/${finalAttrs.version}";
```

File: pkgs/by-name/sr/sratoolkit/package.nix [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+stdenv.mkDerivation (finalAttrs: {
+  pname = "sratoolkit";
+
+  # NOTE: When updating make sure to update ncbi-vdb as well for versions to match
+  version = "3.1.1";
+
+  src = fetchFromGitHub {
+    owner = "ncbi";
+    repo = "sra-tools";
+    rev = finalAttrs.version;
```
Comment (gador) [inline pkgs/by-name/sr/sratoolkit/package.nix:24]:
```suggestion
    rev = "refs/tags/${finalAttrs.version}";
```

File: pkgs/by-name/sr/sratoolkit/package.nix [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    cmake
+    python3
+    bison
+    flex
+    openjdk
+  ];
+
+  meta = {
+    homepage = "https://github.com/ncbi/sra-tools";
+    description = "A collection of tools and libraries for using data in the INSDC Sequence Read Archives";
```
Comment (gador) [inline pkgs/by-name/sr/sratoolkit/package.nix:49]:
```suggestion
    description = "Collection of tools and libraries for using data in the INSDC Sequence Read Archives";
```

**Review by gador — CHANGES_REQUESTED** [review]

**Comment by gador** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 326174` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
2 packages built:
</summary>
<ul>
<li>ncbi-vdb</li>
<li>sratoolkit</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
sratoolkit:
<details>

warning: EvalError
Cannot evaluate attribute ‘sratoolkit’ in ‘/nix/store/czvrsx8xjs9v14x3qy7v63djr1dlladz-nixpkgs’.
</details>
</details>

**Comment by gador** [comment]
Result of `nixpkgs-review pr 326174` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>ncbi-vdb</li>
    <li>sratoolkit</li>
  </ul>
</details>

**Review by gador — APPROVED** [review]

**Pull request merged**